lib/devices/webusb.js

```diff
@@ -0,0 +1,387 @@
+/*!
+ * webusb.js - Ledger Web USB hid communication
+ * Copyright (c) 2019, The Bcoin Developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+/* eslint-env browser */
+
+'use strict';
+
+const assert = require('bsert');
+const {Lock} = require('bmutex');
+
+const Logger = require('blgr');
+const LedgerProtocol = require('../protocol');
+const {LedgerError} = LedgerProtocol;
+const {ProtocolWriter, ProtocolReader} = LedgerProtocol;
+
+const {Device, DeviceInfo} = require('./device');
+
+/**
+ * Ledger WebUSB Packetsize
+ * @const {Number}
+ */
+const PACKET_SIZE = 64;
+
+/**
+ * USB instance
+ * @const {USBDevice}
+ */
+
+const usb = navigator.usb;
```

Comment:
I think its a good idea to check for `navigator` here and stop the process early if it doesn't exist so people get a straight forward error message